transcom/mymove
# [MB-16236] Update go version from 1.20.3 to 1.20.5

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16236)

## Summary

Release Notes: https://go.dev/doc/devel/release#go1.20.minor

[Corresponding circleci-docker repo PR](https://github.com/transcom/circleci-docker/pull/414).

Changes:
- [x] Update go versions in tools files.
- [x] Update docker hashes with a hash that has go 1.20.5. https://hub.docker.com/r/milmove/circleci-docker/tags
- [x] Update and test [`golangci-lint` 1.53.3](https://github.com/golangci/golangci-lint/releases/tag/v1.53.3).  This update caused some new linter failures, so those have been fixed (see the commit history to see those changes in isolation).  The majority were around unused parameters or simplifications of `if` statements.  I also checked for other pre-commit hook updates (via `pre-commit autoupdate`) and updated `markdownlint-cli` as it was the only other one with an update.
- [x] Update `nix` hash once it is available. Nix PR: https://github.com/NixOS/nixpkgs/pull/236359
- [x] Update docker hashes once https://github.com/transcom/circleci-docker/pull/414 merges.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. See that things that use `go` seem to be fine, like running the server and/or backend tests.
